docs/advanced.rst

```diff
@@ -517,39 +519,66 @@ non-bootable system caused by an error or power-loss during the update.
+
+Update eMMC Boot Partitions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+RAUC supports updating eMMC boot partitions (see the JEDEC standard JESD84-B51_
+for details), one of which gets enabled atomically via configuration registers
+in the eMMC (*ext_csd registers*).
 
 .. _JESD84-B51: http://www.jedec.org/standards-documents/results/jesd84-b51
 
-As a further example, the NXP i.MX6 supports up to four bootloader copies when
-booting from NAND flash.
+The required slot type is ``boot-emmc``.
+The device to be specified is expected to be the root device.
+The boot partitions are derived automatically.
+A ``system.conf`` could look like this:
+
+.. code-block:: cfg
+
+  [slot.bootloader.0]
+  device=/dev/mmcblk1
+  type=boot-emmc
+
+.. important::
+
+  A kernel bug may prevent consistent toggling of the eMMC ext_csd boot
+  partition register.
+  Be sure your kernel contains this patch:
+  https://www.spinics.net/lists/linux-mmc/msg48271.html
+
+Bootloader Update Ideas
+~~~~~~~~~~~~~~~~~~~~~~~
+
+The NXP i.MX6 supports up to four bootloader copies when booting from NAND
+flash.
 The ROM code will try each copy in turn until it finds one which is readable
 without uncorrectable ECC errors and has a correct header.
 By using the trait of NAND flash that interrupted writes cause ECC errors and
 writing the first page (containing the header) last, the bootloader images can
 be replaced one after the other, while ensuring that the system will boot even in
 case of a crash or power failure.
 
-Currently, independent of whether you are able to update your bootloader with
-fallback, atomically or with some risk of an unbootable system, our suggestion
-is to handle updates for it outside of RAUC.
-The main reason is to avoid booting an old system with a new bootloader, as this
-combination is usually not tested during development, increasing the risk of
-problems appearing only in the field.
+The slot type could be called "boot-imx6-nand" analogous to eMMC.
+
+Considerations When Updating the Bootloader
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-One possible approach to this is:
+Booting an old system with a new bootloader is usually not tested during
+development, increasing the risk of problems appearing only in the field.
+If you want to address this issue do not add the bootloader to your bundle, but
+rather use an approach like this:
 
 * Store a copy of the bootloader in the rootfs.
-* Use RAUC only to update the rootfs. The combinations to test
+* Use RAUC to update the rootfs. The combinations to test
   can be reduced by limiting which old versions are supported by an update.
 * Reboot into the new system.
 * On boot, before starting the application, check that the current slot
   is 'sane'. Then check if the installed bootloader is older than the
   version shipped in the (new) rootfs. In that case:
 
-  * Disable the old rootfs slot and update the bootloader.
+  * Disable the old rootfs slot and update the bootloader via
```

Comment:
Mh... not sure if this isn't a bit broken. Doesn't write-slot require you to define the slot in the system.conf despite we intentionally do not want to handle this slot with RAUC?..

Reply:
True. Any suggestions? Either define the slot in the system.conf without providing updates in bundles, or write the bootloader without `rauc slot-write`.